lib/crowbar/client/app/upgrade.rb

```diff
@@ -0,0 +1,280 @@
+          `status` will print out a status of the upgrade.
+          You can display the status in different output formats
+          and you can filter the list by any search criteria.
+
+          With --format <format> option you can choose an output format
+          with the available options table, json or plain. You can also
+          use the shortcut options --table, --json or --plain.
+
+          With --filter <filter> option you can limit the result of
+          printed out elements. You can use any substring that is part
+          of the found elements.
+        LONGDESC
+
+        method_option :format,
+          type: :string,
+          default: "table",
+          banner: "<format>",
+          desc: "Format of the output, valid formats are table, json or plain"
+
+        method_option :table,
+          type: :boolean,
+          default: false,
+          aliases: [],
+          desc: "Format output as table, a shortcut for --format table option"
+
+        method_option :json,
+          type: :boolean,
+          default: false,
+          aliases: [],
+          desc: "Format output as table, a shortcut for --format json option"
+
+        method_option :plain,
+          type: :boolean,
+          default: false,
+          aliases: [],
+          desc: "Format output as table, a shortcut for --format plain option"
+
+        method_option :filter,
+          type: :string,
+          default: nil,
+          banner: "<filter>",
+          desc: "Filter by criteria, display only data that contains filter"
+
+        def status
+          Command::Upgrade::Status.new(
+            *command_params
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "prepare",
+          "Prepare Crowbar upgrade"
+
+        long_desc <<-LONGDESC
+          `prepare` will set the nodes to upgrading state
+        LONGDESC
+
+        def prepare
+          Command::Upgrade::Prepare.new(
+            *command_params
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "services",
+          "Stop related services on all nodes during upgrade"
+
+        long_desc <<-LONGDESC
+          `services` will stop all related services on all nodes
+          during the upgrade.
+
+          With --format <format> option you can choose an output format
+          with the available options table, json or plain. You can also
+          use the shortcut options --table, --json or --plain.
+
+          With --filter <filter> option you can limit the result of
+          printed out rows. You can use any substring that is part of
+          the found rows.
+        LONGDESC
+
+        def services
+          Command::Upgrade::Services.new(
+            *command_params
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "prechecks",
+          "Perform checks to make sure Crowbar can be upgraded"
```

Comment:
Upgrading Crowbar means upgrading the whole product, with OpenStack included of course. Otherwise we would have to say "Crowbar server". but I will be more specific in the `long_desc`